src/main/java/seedu/address/model/student/UniqueStudentList.java

```diff
@@ -241,52 +493,166 @@ create a class from the start time and check to see if it exceeds the startTime
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                    // trying to fit a slot from the start of the class
+                    newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                } else if (classToCompare.endTime.plusMinutes(tr.duration).compareTo(tr.endTimeRange) <= 0) {
+                    // fit a class after the end of the class
+                    newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                }
+            } else if (classToCompare.startTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) >= 0) {
+                if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                    newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                }
+            } else if (classToCompare.startTime.compareTo(tr.endTimeRange) >= 0) {
+                newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+            } else {
+                newClass = new Class(currDate.plusDays(1), startTimeFromTr, endTimeFromTr);
+            }
+        } else if (currTime.compareTo(tr.startTimeRange) > 0 && currTime.compareTo(tr.endTimeRange) < 0) {
+            // this means that the currentTime is between the start and end time range
+            if (classToCompare.endTime.compareTo(tr.startTimeRange) <= 0) {
+                if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                    newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                }
+            } else if (classToCompare.endTime.compareTo(tr.startTimeRange) > 0
+                    && classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) <= 0) {
+                if (currTime.compareTo(classToCompare.endTime) <= 0) {
+                    if (endTimeFromClass.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromClass, endTimeFromClass);
+                    }
+                } else {
+                    if (endTimeFromCurrTime.compareTo(tr.endTimeRange) <= 0) {
+                        newClass = new Class(currDate, startTimeFromCurrTime, endTimeFromCurrTime);
+                    }
+                }
+            } else if (classToCompare.endTime.compareTo(tr.endTimeRange) < 0
+                    && classToCompare.startTime.compareTo(tr.startTimeRange) > 0) {
+                if (currTime.compareTo(classToCompare.startTime) < 0) {
+                    if (endTimeFromTr.compareTo(classToCompare.startTime) <= 0) {
+                        newClass = new Class(currDate, startTimeFromTr, endTimeFromTr);
+                    }
```

Comment:
[Missed corner case]
when the current time is before the single class start time, the front part is checked but the part after the class is ignored.
Current time: 2052
Single lesson: 2100-2200
`avail 2030-2359 60` expects to give `2022-11-06 2200-2300` but gives `2022-11-07 2030-2130` in reality.

Reply:
Resolved!